crates/ruff_linter/src/rules/flake8_pytest_style/rules/helpers.rs

```diff
@@ -45,6 +48,66 @@ pub(super) fn is_pytest_parametrize(call: &ExprCall, semantic: &SemanticModel) -
             matches!(qualified_name.segments(), ["pytest", "mark", "parametrize"])
         })
 }
+/// Whether `path` likely leads to a Pytest test file.
+///
+/// Pytest determines a test file using [a number of conventions and settings][1].
+/// Only the conventions are reflected here.
+/// This function returns true if the file name (`.py` notwithstanding):
+///
+/// * Starts with `test_`, or
+/// * Ends with `_test`
+///
+/// It is thus prone to both false positives and false negatives.
+///
+/// [1]: https://docs.pytest.org/en/stable/explanation/goodpractices.html#conventions-for-python-test-discovery
+#[allow(dead_code)]
+pub(crate) fn is_likely_pytest_test_file(path: &Path) -> bool {
+    if !matches!(path.extension().and_then(OsStr::to_str), Some("py")) {
+        return false;
+    }
+
+    let Some(file_stem) = path.file_stem().and_then(OsStr::to_str) else {
+        return false;
+    };
+
+    file_stem.starts_with("test_") || file_stem.ends_with("_test")
+}
+
+/// Whether the currently checked `func` is likely to be a Pytest test.
+///
+/// A normal Pytest test function is one whose name starts with `test` and is either:
+///
+/// * Placed at module-level, or
+/// * Placed within a class whose name starts with `Test` and does not have an `__init__` method.
+///
+/// See also [`is_likely_pytest_test_file`].
+pub(crate) fn is_likely_pytest_test(func: &StmtFunctionDef, checker: &Checker) -> bool {
+    let semantic = checker.semantic();
+
+    if !func.name.starts_with("test") {
```

Comment:
```suggestion
    if !func.name.starts_with("test_") {
```

Reply:
The upstream implementation uses `test_`: https://github.com/m-burst/flake8-pytest-style/blob/92c080312e36c13d2d7725d7bbd89d75d7eaae65/flake8_pytest_style/visitors/t_functions.py#L19

Reply:
We should try to find the relevant code in pytest and then use whatever pytest uses.

Reply:
We can keep `test` because of the default here https://github.com/pytest-dev/pytest/blob/477e9283aabc93f8d3ad34ff0b98f68b5a432aa8/src/_pytest/python.py#L96